packages/workbox-webpack-plugin/src/lib/get-webpack-manifest-entries.js

```diff
@@ -0,0 +1,76 @@
+const url = require('url');
+
+const entry = (revision, filePath, publicPath) => ({
+  revision,
+  url: url.resolve(publicPath, filePath),
+});
+
+/**
+ * Filter chunks that have one of the provided names (from config.chunks)
+ * @param {Array<Object>} chunks Webpack chunks
+ * @param {Array<string>} include Chunk names to include
+ * @param {Array<string>} exclude Chunk names to exclude
+ * @return {Array<Object>} Filtered array of chunks
+ */
+const filterChunks = (chunks, include, exclude) =>
+  chunks.filter((chunk) =>
+    // chunk has a name
+    Object.prototype.hasOwnProperty.call(chunk, 'name')
+      // did not specify include or chunk in include
+      && (!Array.isArray(include) || !!~include.indexOf(chunk.name))
+      // did not specify exclude or chunk not in exclude
+      && (!Array.isArray(exclude) || !~exclude.indexOf(chunk.name))
+  );
+
+/**
+ * Maps webpack asset filenames to their chunk hash
+ * @param {Array<Object>} chunks Webpack chunks
+ * @return {Object} {filename: hash} map
+ */
+const mapAssetsToChunkHash = (chunks) =>
+  chunks.reduce((assetMap, chunk) =>
+      Object.assign(
+        assetMap,
+        ...chunk.files.map((f) => ({[f]: chunk.renderedHash}))
+    ), {});
+
+const generateManifestEntries = (compiler, compilation, config) => {
```

Comment:
Something I noticed when running your code locally was that the `workbox-sw` (and `.map`) file were getting picked up here:

```
  {
    "revision": "9a0110a4f8c1796a1414",
    "url": "/workbox-sw.prod.v2.0.1.js"
  },
  {
    "revision": "9a0110a4f8c1796a1414",
    "url": "/workbox-sw.prod.v2.0.1.js.map"
  },
```

While including these files in the manifest won't break anything (the browser will ignore them when performing `importScripts()`), it means that there's some extra bytes transferred and stored on disk during SW installation, and it would be preferable to exclude them.

Reply:
Oh, I thought we would want to cache these? I am curious, how come we don't want to cache these assets?

Reply:
Anything pulled in with `importScripts` (e.g. `workbox-sw.prod.v2.0.1.js`) is implicitly cached and available offline—this behavior is defined in the service worker specification ([section 7.3.2](https://w3c.github.io/ServiceWorker/#importscripts), item 3.1). Requests for those URLs don't trigger a service worker `fetch` event.

The `.map` isn't covered by this policy, but I don't actually know that the request for the `.map` made by DevTools while debugging actually triggers a `fetch` event either, since DevTools might not be considered an active client of the service worker. And even if the `.map` request does go through the service worker, needing to use the sourcemap while offline is an edge case that wouldn't warrant the extra bytes transferred and stored.

Reply:
> needing to use the sourcemap while offline is an edge case that wouldn't warrant the extra bytes transferred and stored.

I strongly agree with this. I've seen plenty of examples of folks also generating and accidentally caching `.map` files that were pretty large (1MB) causing a perf hit when it's easier to just not cache this asset.